test/test_datasets_utils.py

```diff
@@ -214,5 +216,30 @@ def test_verify_str_arg(self):
         pytest.raises(ValueError, utils.verify_str_arg, "b", ("a",), "arg")
 
 
+class TestFolderUtils:
+    @pytest.mark.parametrize(
+        ("kwargs", "expected_error_msg"),
+        [
+            (dict(is_valid_file=lambda path: pathlib.Path(path).suffix in {".png", ".jpeg"}), "classes c"),
+            (dict(extensions=".png"), r"classes b, c.*?[.]png"),
+            (dict(extensions=[".png", ".jpeg"]), "c.*?[.]png, [.]jpeg"),
```

Comment:
It's not really clear what the actual expected message is

Reply:
`.*?` basically is a non-greedy expression that matches everything until we get a match from a later expression. So without regex syntax, this matches `classes b, c[...].png`. The only other alternative would be to produce the actual part that lies inbetween, for example: `class b, c. Supported extensions are: .png`

Reply:
IMO there's value in making "the parts that lies inbetween" (and before, and after) explicit instead of hiding them in a regex. The documenting value of test is otherwise diminished.